app/src/test/java/de/westnordost/streetcomplete/osm/street_parking/StreetParkingKtTest.kt

```diff
@@ -160,7 +179,125 @@ class StreetParkingTest {
+        verifyAnswer(
+            mapOf(
+                "parking:lane:right" to "parallel",
+                "parking:lane:right:parallel" to "on_kerb",
+                "parking:condition:both" to "customers",
+            ),
+            LeftAndRightStreetParking(
+                StreetParkingProhibited,
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_KERB)
+            ),
+            arrayOf(
+                StringMapEntryAdd("parking:lane:left", "no"),
+                StringMapEntryAdd("parking:condition:left", "no_parking"),
+                StringMapEntryModify("parking:lane:right", "parallel", "parallel"),
+                StringMapEntryModify("parking:lane:right:parallel", "on_kerb", "on_kerb"),
+                StringMapEntryDelete("parking:condition:both", "customers"),
+            )
+        )
+        verifyAnswer(
+            mapOf(
+                "parking:lane:both" to "parallel",
+                "parking:lane:both:parallel" to "on_street",
+                "parking:condition:both" to "customers",
+            ),
+            LeftAndRightStreetParking(
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_STREET),
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_KERB)
+            ),
+            arrayOf(
+                StringMapEntryModify("parking:lane:both", "parallel", "parallel"),
+                StringMapEntryAdd("parking:lane:left:parallel", "on_street"),
+                StringMapEntryAdd("parking:lane:right:parallel", "on_kerb"),
+                StringMapEntryDelete("parking:lane:both:parallel", "on_street"),
+                StringMapEntryDelete("parking:condition:both", "customers"),
+            )
+        )
+    }
+
+    @Test fun `applying removes properties specified for side that changed`() {
+        verifyAnswer(
+            mapOf(
+                "parking:lane:both" to "parallel",
+                "parking:lane:left:parallel" to "on_street",
+                "parking:lane:right:parallel" to "on_kerb",
+                "parking:condition:right" to "customers",
+            ),
+            LeftAndRightStreetParking(
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_STREET),
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_STREET)
+            ),
+            arrayOf(
+                StringMapEntryModify("parking:lane:both", "parallel", "parallel"),
+                StringMapEntryAdd("parking:lane:both:parallel", "on_street"),
+                StringMapEntryDelete("parking:lane:left:parallel", "on_street"),
+                StringMapEntryDelete("parking:lane:right:parallel", "on_kerb"),
+                StringMapEntryDelete("parking:condition:right", "customers")
```

Comment:
So, here I would expect the `parking:condition:right` to be deleted because the right side changed from 
parallel on-kerb parking to parallel on-street parking. (I added that line to the test).

Or am I missing something?